core/src/test/java/feign/ResponseTest.java

```diff
@@ -32,4 +38,40 @@ public void reasonPhraseIsOptional() {
     assertThat(response.reason()).isNull();
     assertThat(response.toString()).isEqualTo("HTTP/1.1 200\n\n");
   }
+
+  @Test
+  public void lowerCasesNamesOfHeaders() {
+    Response response = Response.create(200,
+                                        null,
+                                        Collections.singletonMap("Content-Type",
+                                                                 Collections.singletonList("application/json")),
+                                        new byte[0]);
+    Map.Entry<String, Collection<String>> actualEntry = response.headers().entrySet().iterator().next();
+    Map.Entry<String, Collection<String>> expectedEntry =
+            new AbstractMap.SimpleEntry<>("content-type", Collections.singletonList("application/json"));
+    assertThat(actualEntry).isEqualTo(expectedEntry);
+  }
+
+  @Test
+  public void canAccessHeadersCaseInsensitively() {
+    List<String> valueList = Collections.singletonList("application/json");
+    Response response = Response.create(200,
+                                        null,
+                                        Collections.singletonMap("Content-Type", valueList),
+                                        new byte[0]);
+    assertThat(response.headers().get("content-type")).isEqualTo(valueList);
+    assertThat(response.headers().get("Content-Type")).isEqualTo(valueList);
+  }
+
+  @Test
+  public void headerValuesWithSameNameOnlyVaryingInCaseAreMerged() {
+    Map<String, Collection<String>> headersMap = new LinkedHashMap<>();
+    headersMap.put("Set-Cookie", Lists.newArrayList("Cookie-A=Value", "Cookie-B=Value"));
```

Comment:
nit: using the built-in `Arrays.asList` for things like this is more consistent with other code and.. less chars :P